File: C/Assignment10/Program3.c
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                          //
// File name :   Program3.c                                                                                 //
// Description : Program which accept number from user & return the count of digits in between 3 & 7.       //
// Author :      Pallavi Omprakash Malewar                                                                  //
// Date :        20/06/2025                                                                                 //
//                                                                                                          //
//////////////////////////////////////////////////////////////////////////////////////////////////////////////

/*
Input : 2395
Output : 1

Input : 1018
Output : 0

Input : 4521
Output : 2

Input : 9922
Output : 0
*/

#include<stdio.h>

//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                          //
// Description   : Function accepts integer and returns count of digits which are in between 3 and 7.       //
// Function Name : CountRange                                                                               //
// Input         : int iNo                                                                                  //
// Output        : int (Count of digits between 3 and 7)                                                    //
// Author        : Pallavi Omprakash Malewar                                                                //
// Date          : 20/06/2025                                                                               //
//                                                                                                          //
//////////////////////////////////////////////////////////////////////////////////////////////////////////////

int CountRange(int iNo) //ino defines first number
{
    int iDigit = 0;
    int iCnt = 0;

    while(iNo != 0)

    {
        iDigit = iNo % 10;

        if((iDigit >= 3) && (iDigit <= 7))
        {
            iCnt++;
        }
        iNo = iNo / 10;
    }

    return iCnt;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = CountRange(iValue);

    printf("Total Digits between 3 & 7 : %d\n",iRet);

    return 0;
}

File: JAVA/Assignment10/Program3.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a program which accept number from user and return the    //
//                     count of digits in between 3 & 7.                               //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 30 Jun 2026                                                                  //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////

/*

    Input: 2395
    Output : 1

    Input: 1018
    Output : 0

    Input: 4521
    Output : 2

    Input: 9922
    Output : 0

*/


import java.util.Scanner;

class Program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Number : ");
        int No = sobj.nextInt();

        int Count = CountOdd(No);

        System.out.println(Count);
    }

    public static int CountOdd(int No)
    {
        if(No == 0) return 0;
        int iCount = 0, iDigit = 0;

        while(No != 0)
        {
            iDigit = No % 10;
            if(iDigit > 3 && iDigit < 7)
            {
                iCount++;
            }
            No /= 10;
        }
        return iCount;
    }
}